### PR TITLE
Use snapkit to layout datePickerView

### PR DESCRIPTION
**Pull request by Omarrasheed**

### CODE REVIEW

File: TCAT/Views/DatePickerView.swift [hunk 1]
```diff
@@ -61,177 +61,139 @@ class DatePickerView: UIView {
...
-        timeTypeSegmentedControl.frame = CGRect(x: 0, y: cancelButton.frame.maxY + spaceBtButtonAndSegmentedControl, width: (self.frame.width*(343/375) - spaceBtSegmentControls)*(2/3), height: segmentedControlHeight)
+    // MARK: Set data
 
-        timeTypeSegmentedControl.center.x = self.frame.width/2 + timeTypeSegmentedControl.frame.width/4 + spaceBtSegmentControls/2
-    }
+    private func setupConstraints() {
+        cancelButton.snp.makeConstraints { make in
+            make.leading.equalToSuperview().inset(spaceBtButtonAndSuperviewSide)
+            make.top.equalToSuperview().inset(spaceBtButtonAndSuprviewTop)
+            make.size.equalTo(CGSize(width: 60, height: buttonHeight))
```
Comment (kevinchan159) [inline TCAT/Views/DatePickerView.swift]: move numbers into constants at top of function

File: TCAT/Views/DatePickerView.swift [hunk 2]
```diff
@@ -251,4 +213,25 @@ class DatePickerView: UIView {
...
+
+    @objc private func leaveNowSegmentedControlValueChanged(segmentControl: UISegmentedControl) {
+        datepicker.date = Date()
+    }
+
+    // MARK: Datepicker
+
+    @objc private func datepickerValueChanged(datepicker: UIDatePicker) {
+        if Time.compare(date1: datepicker.date, date2: Date()) == ComparisonResult.orderedSame {
+            leaveNowSegmentedControl.selectedSegmentIndex = typeToSegmentControlElements[.leaveNow]!.index
```
Comment (kevinchan159) [inline TCAT/Views/DatePickerView.swift]: can we unwrap instead of `!`

File: TCAT/Views/DatePickerView.swift [hunk 3]
```diff
@@ -251,4 +213,25 @@ class DatePickerView: UIView {
         delegate?.dismissDatePicker()
     }
 
+    // MARK: Segement Controls
+
+    @objc private func timeTypeSegmentedControlValueChanged(segmentControl: UISegmentedControl) {
+        if timeTypeSegmentedControl.selectedSegmentIndex == typeToSegmentControlElements[.arriveBy]!.index {
```
Comment (kevinchan159) [inline TCAT/Views/DatePickerView.swift]: unwrap instead of `!`

File: TCAT/Views/DatePickerView.swift [hunk 4]
```diff
@@ -61,177 +61,139 @@ class DatePickerView: UIView {
...
     }
 
-    func setDatepickerTimeType(searchTimeType: SearchType) {
-        switch searchTimeType {
-        case .leaveAt, .leaveNow:
-            timeTypeSegmentedControl.selectedSegmentIndex = typeToSegmentControlElements[.leaveAt]!.index
-        case .arriveBy:
-            timeTypeSegmentedControl.selectedSegmentIndex = typeToSegmentControlElements[.arriveBy]!.index
+    private func setSegmentedControlOptions(_ segmentedContol: UISegmentedControl, options: [String]) {
+        for i in options.indices {
```
Comment (kevinchan159) [inline TCAT/Views/DatePickerView.swift]: pref to use `forEach`

**Review by kevinchan159 — CHANGES_REQUESTED** [review]
nice just left few comments

**Comment by Omarrasheed** [comment]
Updated! @kevinchan159

**Review by kevinchan159 — APPROVED** [review]
NOICE, just remember to squash before merging

File: TCAT/Views/DatePickerView.swift [hunk 5]
```diff
@@ -251,4 +201,26 @@ class DatePickerView: UIView {
         delegate?.dismissDatePicker()
     }
 
+    // MARK: Segement Controls
```
Comment (young-k) [inline TCAT/Views/DatePickerView.swift]: You spelled "segment" wrong!

**Review by young-k — APPROVED** [review]
Fix & feel free to merge afterwards

**Pull request merged**